plugins/modules/host.py

```diff
@@ -0,0 +1,284 @@
+        rack_id:
+            description: The rack ID for this host.
+            type: str
+            returned: optional
+        role_refs:
+            description: The list of roles assigned to this host.
+            type: list
+            returned: optional
+        tags:
+            description: Tags associated with the host.
+            type: list
+            returned: optional
+        total_phys_mem_bytes:
+            description: he amount of physical RAM on this host, in bytes.
+            type: str
+            returned: optional
+"""
+
+
+class ClouderaHost(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClouderaHost, self).__init__(module)
+
+        # Initialize the return values
+        self.host_name = self.get_param("host_name")
+        self.cluster_name = self.get_param("cluster_name")
+        self.host_ip = self.get_param("host_ip")
+        self.state = self.get_param("state")
+        # Execute the logic
+        self.process()
+
+    def check_required_host_id_param(self):
+        if self.host_ip is None:
+            error_msg = "host_ip"
+            self.module.fail_json(msg=f"Missing required parameter: {error_msg}")
+
+    def check_required_cluster_name_param(self):
+        if self.cluster_name is None:
+            error_msg = "cluster_name"
+            self.module.fail_json(msg=f"Missing required parameter: {error_msg}")
+
+    @ClouderaManagerModule.handle_process
+    def process(self):
+
+        cluster_api_instance = ClustersResourceApi(self.api_client)
+        host_api_instance = HostsResourceApi(self.api_client)
+        self.host_output = {}
+        self.changed = False
+        existing = None
+
+        try:
+            hosts = host_api_instance.read_hosts().to_dict()
+            for host in hosts['items']:
+                if self.host_name == host['hostname']:
+                    host_id = host['host_id']
+                    existing = host_api_instance.read_host(host_id=host_id).to_dict()
+                    break
+                
+        except ApiException as ex:
+            if ex.status != 404:
+                raise ex  
+            
+        if self.state in ['present']:
+            if existing:
+                self.host_output = existing
+            else:
+                self.check_required_host_id_param()
+                body = {"items": [{"hostname": self.host_name,"ipAddress": self.host_ip }]}
+                create_host = host_api_instance.create_hosts(body=body).to_dict()
+                self.host_output = host_api_instance.read_host(host_id=create_host['items'][0]['host_id']).to_dict()
+                self.changed = True
+
+        if self.state in ['absent']:
+            if existing:
+                self.host_output = host_api_instance.delete_host(host_id=existing['host_id']).to_dict()
+                self.changed = True
+
+        if self.state in ['attach','detach']:
+            self.check_required_cluster_name_param()
```

Comment:
This logic can be pushed into the `argument_spec` using [required_if](https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#dependencies-between-module-options)